src/runtime/cuda/cuda_queue.cpp

```diff
@@ -47,12 +47,61 @@
 
 #include <cassert>
 #include <memory>
+#include <vector>
 
 namespace hipsycl {
 namespace rt {
 
 namespace {
 
+
+unsigned select_ptx_version(unsigned sm_version, unsigned& ptx_target) {
+  // Our bitcide libraries need at least ptx +60
+  unsigned ptx_version = 60;
+  
+  // for each sm, stores minimum ptx version required
+  // based on data from
+  // https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#ptx-module-directives-target
+  static std::vector<std::array<unsigned, 2>> sm_min_ptx_version =
+      {{72, 61},
+       {75, 63},
+       {80, 70},
+       {86, 71},
+       {87, 74},
+       {89, 78},
+       {90, 78},
+       // some variants of 100, 101 need 8.6, some 8.8
+       // so require 8.8 for all to be on the safe side.
+       {100, 88},
+       {101, 88},
+       {103, 88},
+       {120, 87},
```

Comment:
If we care about 120f (which does not make much sense to choose for JIT, I guess, but for consistency with 100f):

```suggestion
       {120, 88},
```

Reply:
Thanks! feels like NVIDIA learnt from AMD here with this SM variant stuff with different requirements :/